### PR TITLE
Raise AuthenticationError 

### PR DESCRIPTION
**Pull request by Darcy-Linde**

### CODE REVIEW

File: polygon/websocket/__init__.py [hunk 1]
```diff
@@ -97,9 +98,11 @@ async def connect(
...
                 if self.verbose:
                     print("authed:", auth_msg)
-                if auth_msg_parsed[0]["status"] == "auth_failed":
-                    print(auth_msg_parsed[0]["message"])
-                    return
+                try:
+                    if auth_msg_parsed[0]["status"] == "auth_failed":
+                        raise AuthenticationError(auth_msg_parsed[0]["message"])
+                except (AuthenticationError):
+                    exit("Authentication Failed. Exiting...")
```
Comment (clickingbuttons) [inline polygon/websocket/__init__.py:105]: Remove this try/except. Let the user handle the error.

File: polygon/websocket/__init__.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import inspect
 import ssl
 import certifi
+from multiprocessing import AuthenticationError
```
Comment (clickingbuttons) [inline polygon/websocket/__init__.py:8]: Make our own instead of importing from `multiprocessing` (whatever that is)

**Review by clickingbuttons — CHANGES_REQUESTED** [review]
Also add to the `connect` pydoc `:raises AuthenticationError: If invalid API key is supplied.`

File: polygon/websocket/__init__.py (hunk 2 above)
Comment (Darcy-Linde) [reply polygon/websocket/__init__.py:8]: Should I just add the new class at the top of the file or should it go somewhere else?

Comment (clickingbuttons) [reply polygon/websocket/__init__.py:8]: New class at the top is fine.

**Review by clickingbuttons — APPROVED** [review]
Ty!

**Pull request merged**